dask_image/ndmeasure/_utils/_find_objects.py

```diff
@@ -0,0 +1,57 @@
+import numpy as np
+import pandas as pd
+from dask.delayed import delayed, Delayed
+import dask.dataframe as dd
+
+
+def _array_chunk_location(block_id, chunks):
+    """Pixel coordinate of top left corner of the array chunk."""
+    array_location = []
+    for idx, chunk in zip(block_id, chunks):
+        array_location.append(sum(chunk[:idx]))
+    return tuple(array_location)
+
+
+@delayed
+def _find_bounding_boxes(x, array_location):
+    """An alternative to scipy.ndi.find_objects"""
+    unique_vals = np.unique(x)
+    unique_vals = unique_vals[unique_vals != 0]
+    result = {}
+    for val in unique_vals:
+        positions = np.where(x == val)
+        slices = tuple(slice(np.min(pos) + array_location[i], np.max(pos) + 1 + array_location[i], 1) for i, pos in enumerate(positions))
+        result[val] = slices
+    return pd.DataFrame.from_dict(result, orient='index')
+
+
+def _combine_slices(slices):
+    "Return the union of all slices."
+    if len(slices) == 1:
+        return slices[0]
+    else:
+        start = min([sl.start for sl in slices])
+        stop = max([sl.stop for sl in slices])
+        return slice(start, stop, 1)
+
+
+def _merge_bounding_boxes(x, ndim):
+    x = x.dropna()
+    data = {}
+    for i in range(ndim):
+        slices = [x[ii] for ii in x.index if str(ii).startswith(str(i))]
+        combined_slices = _combine_slices(slices)
+        data[i] = combined_slices
+    result = pd.Series(data=data, index=[i for i in range(ndim)], name=x.name)
+    return result
+
+
+def _find_objects(df1, df2, ndim=2):
```

Comment:
Still need to figure out how I can pass in a keyword argument for either `ndim`/`meta` (probably more efficient to just pass `meta` in rather than continually recompute it). It doesn't seem like this is possible to do with dask bag fold?

I might be able to figure out a workaround by counting all the dataframe column headings with some regex, but I'd prefer not to do that.

Reply:
Martin has also suggested I try using functools partial, which sounds like a good approach

Reply:
Martin's suggestion of functools partial is working really well :)